components/raftstore/src/store/worker/pd.rs

```diff
@@ -1918,6 +1925,17 @@ where
         self.slow_score.last_tick_id += 1;
         self.slow_score.last_tick_finished = false;
         if self.slow_score.last_tick_id % self.slow_score.round_ticks == 0 {
+            // `last_update_time` is refreshed every round. We use it to maintain
+            // the health status of the TiKV.
+            let health_status =
+                if self.slow_score.last_record_time < self.slow_score.last_update_time {
+                    ServingStatus::ServiceUnknown
+                } else {
+                    ServingStatus::Serving
+                };
+            if let Some(health_service) = &self.health_service {
+                health_service.set_serving_status("", health_status);
```

Comment:
How about updating the health status according to the slow score or the ratio of timeout records? For instance, if the ratio of timeout records exceeds `ratio_thresh`, then set the status as `ServiceUnknown` as the same as the way we update the slow score.

Reply:
I'm inclined to be conservative. This PR only intends to solve the problem when the raftstore is totally stuck instead of just being slow.

Because being slow does not mean the leader will certainly transfer, setting the status to unknown too easily will cause unexpected request forwarding.

And if the leader transfers due to just slowness, the TiKV can still return `NotLeader` to the client. It does not need the involvement of the health status.